[otbn] Various specificaton cleanups

imphil opened the pull request:
See individual commit messages for details.

rswarbrick commented on hw/ip/otbn/doc/_index.md:
```diff
@@ -46,7 +46,7 @@ The instruction set is split into two groups:
 * The **base instruction subset** operates on the 32b General Purpose Registers (GPRs).
   Its instructions are used for the control flow of a OTBN application.
-  The base instructions are inspired by RISC-V’s RV32I instruction set, but not compatible with it.
+  The base instructions are inspired by RISC-V's RV32I instruction set, but not compatible with it.
```
Can you change the commit message to **non**-ASCII character? :-)

imphil replied:
Uh, updated.

rswarbrick commented on hw/ip/otbn/doc/_index.md:
I'm not sure that "Privilege" is the right word here. Just "Access", maybe?

imphil replied:
I agree and changed to "Access". "Privilege" as column title came from the RISC-V Privileged spec, which also has the privilege mode in this column (e.g. "URW" for R/W from U mode).

imphil commented:
Updated to address your review comments @rswarbrick. I also added another commit that fixes the WSR documentation for ACC and URND, which I somehow messed up a couple days earlier.

rswarbrick reviewed (approved):
Looks great!
